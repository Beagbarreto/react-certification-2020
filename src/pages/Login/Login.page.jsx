import React, {useState, useEffect} from 'react';
import { useHistory } from 'react-router';

import { useAuth } from '../../providers/Auth';
import './Login.styles.css';

function LoginPage() {
  const { login, authenticated, error } = useAuth();
  const history = useHistory();

  useEffect(() => {
    if (authenticated) {
      history.push('/');
    }
  }, [authenticated, history]);

  function authenticate(e) {
    e.preventDefault();
    login(username, password);
  }

  const [username, setUsername] = useState();
  const [password, setPassword] = useState();

  return (
    <section className="login">
      <h3>Welcome back, login!</h3>
      <form onSubmit={authenticate} className="login-form">
        {error && <span className="error-text">{error}</span>}
        <div className="form-group">
          <label htmlFor="username">
            <strong>username </strong>
            <input
              required type="text"
              id="username"
              onChange={e => setUsername(e.currentTarget.value)}
            />
          </label>
        </div>
        <div className="form-group">
          <label htmlFor="password">
            <strong>password </strong>
            <input
              required
              type="password"
              id="password"
              onChange={e => setPassword(e.currentTarget.value)}
            />
          </label>
        </div>
        <button type="submit">login</button>
      </form>
    </section>
  );
}

export default LoginPage;
