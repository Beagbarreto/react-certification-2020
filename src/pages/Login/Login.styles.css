.login {
  width: 350px;
}

.login h1 {
  text-align: center;
  letter-spacing: -1px;
  color: #E92C8A;
}

.login h3, h5 {
  text-align: center;
  letter-spacing: -1px;
  color: #D4AF37;
}

.login-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.form-group {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
}

.form-group strong {
  display: block;
  font-weight: 700;
  text-transform: capitalize;
  margin-bottom: 0.4rem;
  color: #432272;
}

.form-group input {
  color: white;
  font-size: 1.2rem;
  width: 100%;
  padding: 0.4rem 0.6rem;
  border-radius: 3px;
  border: 1px solid white;
  background-color: rgba(0, 0, 0, 0.1);
}

.login-form button[type='submit'] {
  width: 5rem;
  margin-top: 1rem;
  padding: 0.4rem 0.6rem;
  font-size: 1.2rem;
  border: none;
  border-radius: 3px;
}

.error-text{
  color:#ff3d41;
}