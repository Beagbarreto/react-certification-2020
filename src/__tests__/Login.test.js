import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import AuthProvider from '../providers/Auth/Auth.provider';
import Login from '../pages/Login';

describe('Login Component', () => {
  it('Should render Login', () => {
      const login = render(<Login />);
      expect(login).toMatchSnapshot();
    
      const user = screen.getByLabelTest('username');
      expect(user).toBeInTheDocument();
      const password = screen.getByLabelText('password');
      expect(password).toBeInTheDocument();
    });

  it('should take a snapshot', () => {
    const { asFragment } = render(
      <AuthProvider>
        <SearchContextProvider>
          <Login />
        </SearchContextProvider>
    </AuthProvider>
    );
    expect(
      asFragment(
        <AuthProvider>
          <SearchContextProvider>
            <Login />
          </SearchContextProvider>
        </AuthProvider>
      )
    ).toMatchSnapshot();
  });
});
