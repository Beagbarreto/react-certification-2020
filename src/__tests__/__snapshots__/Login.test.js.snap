// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="login"
  >
    <h3>
      Welcome back, login!
    </h3>
    <form
      class="login-form"
    >
      <div
        class="form-group"
      >
        <label
          for="username"
        >
          <strong>
            username 
          </strong>
          <input
            id="username"
            required=""
            type="text"
          />
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="password"
        >
          <strong>
            password 
          </strong>
          <input
            id="password"
            required=""
            type="password"
          />
        </label>
      </div>
      <button
        type="submit"
      >
        login
      </button>
    </form>
  </section>
</DocumentFragment>
`;
